crates/symbolicator/src/endpoints/applecrashreport.rs

```diff
@@ -11,6 +15,31 @@ use crate::utils::sentry::ConfigureScope;
 
 use super::ResponseError;
 
+struct MultipartField<'a> {
+    inner: axum::extract::multipart::Field<'a>,
+}
+
+impl<'a> MultipartField<'a> {
+    fn new(inner: axum::extract::multipart::Field<'a>) -> Self {
+        Self { inner }
+    }
+}
+
+impl<'a> Stream for MultipartField<'a> {
+    type Item = Result<bytes::Bytes, std::io::Error>;
+
+    fn poll_next(
+        mut self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        let this = &mut self;
+        let inner = std::pin::Pin::new(&mut this.inner);
```

Comment:
I’m a bit surprised. I was using an unsafe block everywhere to wrap futures, but it can be totally safe as well?

Reply:
"safe" in as far pin projection is safe.  but yes, no unsafe block needed, though every time i need this i need to go look up examples of what the trick is.  And lol, just noticed I missed a deref, it should be `let this = &mut *self` so that `this` is no longer pinned.  But according to rust analyser it might not matter because the type of inner doesn't change...